libraries/AP_Scheduler/AP_Scheduler.h

```diff
@@ -21,20 +21,26 @@
  */
 #pragma once
 
-#include <AP_Param/AP_Param.h>
+#include <stdint.h>
+#include <stdio.h>
+
 #include <AP_HAL/Util.h>
+#include <AP_HAL/AP_HAL.h>
+#include <AP_Param/AP_Param.h>
+#include <AP_Vehicle/AP_Vehicle.h>
 
-#define AP_SCHEDULER_NAME_INITIALIZER(_name) .name = #_name,
 
-/*
-  useful macro for creating scheduler task table
- */
-#define SCHED_TASK_CLASS(classname, classptr, func, _rate_hz, _max_time_micros) { \
-    .function = FUNCTOR_BIND(classptr, &classname::func, void),\
-    AP_SCHEDULER_NAME_INITIALIZER(func)\
-    .rate_hz = _rate_hz,\
-    .max_time_micros = _max_time_micros\
-}
+// return the input parameter as a string
+#define xstr(s) str(s)
+#define str(s) #s
+
+#if APM_BUILD_TYPE(APM_BUILD_ArduCopter)
```

Comment:
Not allowed to have APM_BUILD_TYPE in headers anymore.

@guludo This wasn't detected, I think because this header was included only in vehicles and we only check library included headers?